java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -1271,6 +1276,11 @@ public static boolean checkNormalize(MessageBytes uriMB) {
             }
         }
 
+        // The URL must start with '/'
+        if (c[start] != '/') {
+            return false;
+        }
+
         // Check for ending with "/." or "/.."
```

Comment:
The same goes for this test.

Reply:
I do have an input that triggers a 500 response if that check is removed:
```
00000000: 202e 2e20 3020 0a                         .. 0 .
```

That input trigger the exception that's in the PR description.

Reply:
I don't see how that can trigger a 500. The request line:
`GET  %20%2e%2e%20%30%20%0a HTTP/1.1`
is going to result in a 400 response in all current Tomcat versions.
Please provide a request line example that triggers the exception in the original report.

Reply:
Sorry for the confusion. What I posted was the entire query, i.e. you can reproduce the issue with:
```
$ echo -ne "\x20\x2e\x2e\x20\x30\x20\x0a" | nc localhost 8081
HTTP/1.1 500
Date: Tue, 02 Jul 2019 13:55:20 GMT
Connection: close
```

Looking more into it, you can also be reproduced with a normal looking HTTP query:
```
$ echo -ne "GET .. HTTP/1.1\r\n\r\n" | nc localhost 8081
```

Reply:
Thanks. That is what I needed. That should be caught and rejected earlier. I wonder what is going wrong.

Reply:
I was surprised to see that `checkNormalize` was still called when `normalize` returns false in postParseRequest. Maybe that'd be a better fix?

Reply:
Agreed. I reached the same conclusion. I'll get that fixed shortly. Thanks for highlighting this.